### PR TITLE
fix list out of view for upkey and downKey

### LINKED ISSUE #7323: List - when using keyboard, active list item isn't always in view
### Expected Behavior

When using arrow keys to get to list item, would expect active list item to be in view on the screen.

### Actual Behavior

Active list item can end up being off screen but page doesn't scroll down.

### URL, screen shot, or Codepen exhibiting the issue

https://storybook.grommet.io/?path=/story/visualizations-list-onclickitem--on-click-item-list

https://github.com/user-attachments/assets/b5ca4265-db00-4132-8cda-89b263866b01

### Steps to Reproduce

1. Make window smaller than length of list
2. Tab to List
3. Use arrow keys to move down

### Your Environment

- Grommet version: Latest (on stable of storybook)
- Browser Name and version: Chrome
- Operating System and version (desktop or mobile): MacOS

### PR DESCRIPTION
**Pull request by Sulaymon333**
#### What does this PR do?
- Fixes list out of view for up key and down key for List component

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
closes #7323

### CODE REVIEW

File: src/js/components/List/List.js [hunk 1]
```diff
@@ -315,7 +315,20 @@ const List = React.forwardRef(
               event.preventDefault();
               if (active) {
                 const min = onOrder ? 1 : 0;
-                updateActive(Math.max(active - 1, min));
+                const activeElementIndex = Math.max(active - 1, min);
+                updateActive(activeElementIndex);
+
+                // Ensure the active item is in view
+                // setTimeout for activeElement to be updated
+                setTimeout(() => {
```
Comment (jcfilben) [inline src/js/components/List/List.js:323]: Do we need the timeout? I tested without it and wasn't seeing a difference in behavior

Comment (Sulaymon333) [reply src/js/components/List/List.js:323]: Indeed the behavior is the same without the timeout, timeout is implemented to guarantee that the activeElement is updated  prior to executing any operations on it and eventually it ensures that all test cases pass. I have shown a screenshot where the timeout is commented and the test case is failing.

I hope this clarifies the need for the timeout. Looking through the code base, a similar approach is employed for example, in the SkipLinks component, Layer component, and some other places.

image

Comment (jcfilben) [reply src/js/components/List/List.js:323]: Thanks for the info!

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by MikeKingdom — APPROVED** [review]

**Pull request merged**